end_to_end_tests/end_to_end_test_helpers.py

```diff
@@ -0,0 +1,203 @@
+import importlib
+import os
+import shutil
+from filecmp import cmpfiles, dircmp
+from pathlib import Path
+import sys
+import tempfile
+from typing import Any, Callable, Dict, Generator, List, Optional, Set, Tuple
+
+from attrs import define
+import pytest
+from click.testing import Result
+from typer.testing import CliRunner
+
+from openapi_python_client.cli import app
+from openapi_python_client.utils import snake_case
+
+
+@define
+class GeneratedClientContext:
+    """A context manager with helpers for tests that run against generated client code.
+    
+    On entering this context, sys.path is changed to include the root directory of the
+    generated code, so its modules can be imported. On exit, the original sys.path is
+    restored, and any modules that were loaded within the context are removed.
```

Comment:
Btw, the proof that this works is simply that running all the tests in `generated_code_live_tests` in a single pytest session consistently works. Since many of the test classes reuse the same modules and class names for different things, if the sandboxing didn't work then the tests would inevitably pollute each other and fail.